module/x/gravity/keeper/keeper.go

```diff
@@ -165,6 +178,59 @@ func (k Keeper) iterateEthereumSignatures(ctx sdk.Context, storeIndex []byte, cb
 	}
 }
 
+func (k Keeper) IterateBatchTxEthereumSignatures(ctx sdk.Context, cb func(common.Address, uint64, sdk.ValAddress, []byte) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, []byte{types.EthereumSignatureKey, types.BatchTxPrefixByte})
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		key := bytes.NewBuffer(iter.Key())
+		key.Next(2) // trim prefix bytes
```

Comment:
If this is working correctly in tests, I'm surprised -- why is the prefix 2 bytes? The iota we're using should only take one.

Reply:
The iterator is built with two prefix bytes a few lines above:

```go
iter := sdk.KVStorePrefixIterator(store, []byte{types.EthereumSignatureKey, types.BatchTxPrefixByte})
```